community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/ImportLogic.java

```diff
@@ -157,16 +157,35 @@ public void initialize( Input input ) throws IOException
         idMapper = input.idMapper( numberArrayFactory );
         idGenerator = input.idGenerator();
         nodeRelationshipCache = new NodeRelationshipCache( numberArrayFactory, config.denseNodeThreshold() );
-        memoryUsageStats = new MemoryUsageStatsProvider( nodeRelationshipCache, idMapper, neoStore );
         nodes = input.nodes();
         relationships = input.relationships();
         cachedNodes = cachedForSure( nodes, inputCache.nodes( MAIN, true ) );
-        dependencies.satisfyDependencies( input.calculateEstimates( neoStore.getPropertyStore().newValueEncodedSizeCalculator() ),
-                idMapper, neoStore, nodeRelationshipCache );
+        Estimates inputEstimates = input.calculateEstimates( neoStore.getPropertyStore().newValueEncodedSizeCalculator() );
+        sanityCheckEstimatesWithRecordFormat( inputEstimates );
```

Comment:
would really like to see tests for those sanity checks

Reply:
Pulled it out and unit tested that, looks better?